docs/DeveloperGuide.md

```diff
@@ -935,11 +941,70 @@ tutorial group as XLSX files. The resulting XLSX file will either contain attend
 each student across all tutorial lessons in the tutorial group. These features were designed with the user's needs in
 mind after surveying all the CS2101 tutors and interviewing a CS2101 tutor.
 
-### **G.2. Conclusion**
+### **G.2. Challenges**
+
+Throughout the development of **Serenity**, significant hurdles were faced in development.
+We detail some of the challenges faced, and steps we took to solve them.
+
+**Challenge 1: Design stage**
+
+The first challenge we faced involved the planning stage, where we had to have various
+different entities (tutorial groups, lessons, students and their records) come together.
+Compared to **AddressBook**, which had a single `Person` object in the model, we had to
```

Comment:
You can replace all "AddressBook" with "Address Book 3" for standardization purposes